src/main/java/at/hannibal2/skyhanni/utils/renderables/Renderable.kt

```diff
@@ -45,6 +46,24 @@ import java.awt.Color
 import java.util.Collections
 import kotlin.math.max
 
+class Searchable(val renderable: Renderable, val string: String?)
+
+fun Renderable.toSearchable(searchText: String? = null) = Searchable(this, searchText?.removeColor())
+fun Searchable.toRenderable() = renderable
+fun List<Searchable>.toRenderable() = map { it.toRenderable() }
+fun List<Searchable>.toMap() = associate { it.renderable to it.string }
+fun List<Searchable>.buildSearchBox(): Renderable {
+    val textInput = TextInput()
+    val key = 0
+    return Renderable.searchBox(
+        Renderable.verticalSearchableContainer(toMap(), textInput = textInput, key = key + 1),
+        "Search: ",
+        onUpdateSize = { println("onUpdateSize") },
+        textInput = textInput,
+        key = key,
+    )
+}
```

Comment:
Maybe move this into it's own file, since Renderable is already huge (and will grow even more) but Searchable can be extracted into another file.

Reply:
already done. see commit further down